### ISSUE TITLE
No value for column when attempting to insert a record without providing a value for a column with a default

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#6910]
Customer reported this but doesn't have a specific repro.

Other tidbit of info was they were using Laravel and set the default column through there. It was on a column added to a table, not present on the table at creation time.

```php
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::table('testimonials', function (Blueprint $table) {
            $table->boolean('favorite')->default(false)->after('video');
        });
    }
```

Will attempt a few repro cases.

**Comment by jennifersp** [comment]
[This change](https://github.com/dolthub/go-mysql-server/pull/2119/files#diff-690856e48efe2c441527c6213e2507d81f32acb13c3a735216c35d735508a30eL1023) may be related to this issue. This was a fix for column default value gets lost when running an engine test using server as a prepare statement.

**Comment by sbrow** [comment]
Seems to be solved in 1.24.2

**Comment by fulghum** [comment]
Thanks for confirming that things are working correctly in 1.24.2. We'll go ahead and close this issue, but please don't hesitate to let us know if you run into any other problems and we'll be happy to help!
